js/api/Submissions.js

```diff
@@ -0,0 +1,210 @@
+import Parse from 'parse/react-native'
+import realm from '../data/Realm'
+import {currentUser} from '../api/Account'
+import crypto from 'crypto-js'
+import async from 'async'
+
+const Submission = Parse.Object.extend('Submission');
+
+// Fetches the cached submissions related to a specific form
+export function loadCachedSubmissions(formId) {
+  return realm
+    .objects('Submission')
+    .filtered(`formId = "${formId}"`)
+    .sorted('created');
+}
+
+/**
+ * user may only have one submission for each formId
+ *
+ * @param {string} formId, the unique id for the parse form record
+ * @param {object} currentUser, the current parse user saved to AsyncStorage
+ */
+function genSubmissionId(formId, currentUser, done) {
+  const str = '' + currentUser.id + formId;
+  const id = crypto.MD5(str).toString();
+  done(null, id);
+};
+
+/**
+ * create a submission on parse-server
+ *
+ * @param {string} id, the unique id for the realm record
+ * @param {string} formId, the unique id for the parse form record
+ * @param {object} answers, the answers to the current form
+ * @param {object} currentUser, the current parse user saved to AsyncStorage
+ */
+function createParseSubmission(id, formId, answers, currentUser, done) {
+  let submission = new Submission();
+  submission.set('uniqueId', id);
+  submission.set('formId', formId);
+  submission.set('answers', answers);
+  submission.setACL(new Parse.ACL(currentUser));
```

Comment:
The uniqueId is md5 hash of user + formId.  This will enforce that only one submission per form per user.  The objectId on parse is their own UUID that is set upon save.